Source/Widgets/Viewer/Viewer.js

```diff
@@ -141,6 +142,15 @@ function trackDataSourceClock(timeline, clock, dataSource) {
     if (defined(dataSourceClock)) {
       dataSourceClock.getValue(clock);
       if (defined(timeline)) {
+        if (
+          JulianDate.equals(dataSourceClock.startTime, dataSourceClock.stopTime)
+        ) {
+          JulianDate.addSeconds(
+            dataSourceClock.startTime,
+            5,
+            dataSourceClock.stopTime
+          );
+        }
```

Comment:
It feels a little wrong to modify the data sources's `stopTime` when it's really just the timeline that needs special behavior. This is a good place to use a scratch variable.

```js
var scratchStopTime = new JulianDate();

function trackDataSourceClock(timeline, clock, dataSource) {
  if (defined(dataSource)) {
    var dataSourceClock = dataSource.clock;
    if (defined(dataSourceClock)) {
      dataSourceClock.getValue(clock);
      if (defined(timeline)) {
        var startTime = dataSourceClock.startTime;
        var stopTime = dataSourceClock.stopTime;
        if (JulianDate.equals(startTime, stopTime)) {
          stopTime = JulianDate.addSeconds(
            dataSourceClock.startTime,
            CesiumMath.EPSILON2,
            scratchStopTime
          );
        }
        timeline.updateFromClock();
        timeline.zoomTo(startTime, stopTime);
      }
    }
  }
}
```